packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -2223,10 +2231,9 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
   }
 
   Widget _buildMinutePicker(EdgeInsetsDirectional additionalPadding, Widget selectionOverlay) {
+    minuteController=FixedExtentScrollController(initialItem: selectedMinute ~/ widget.minuteInterval);
```

Comment:
It is bug that controller is created on every build. 
It should be created in initState.
And it should be updated in didUpdateWidget in case minuteInterval changed.

The same relates to other controllers.

And, please, add spaces around `=`.

Reply:
Done ◡̈

Reply:
Looks good for me. @TahaTesser , it seems you have more context than me. Could you review please?

Reply:
Tests started failing. It seems something is missed.

Reply:
You can convert your PR to draft, while debugging the tests, to show you are in process and the PR is not ready for review yet.

Reply:
Fixed tests, all checks are passing now, can you please re-review it?